### PR TITLE
fix parseISA one missing element panic

### LINKED ISSUE #4: parseISA panics if it is exactly one element short.
https://github.com/tmc/x12/blob/6f99d45ab2c4fac9489c681d2f79bb811bf6b4e7/decode.go#L212
The number of elements in a valid ISA is 17. When the number of elements is exactly 1 short, this triggers a panic on:
https://github.com/tmc/x12/blob/6f99d45ab2c4fac9489c681d2f79bb811bf6b4e7/decode.go#L232

### PR DESCRIPTION
**Pull request by apxamccallum**
Fixes #4

### CODE REVIEW

File: x12_test.go [hunk 1]
```diff
@@ -144,6 +161,9 @@ IEA*1*000095071~`,
 				t.Errorf("Decode() error = %v, wantErr %v", err, tt.wantErr)
 				return
 			}
+			if err != nil {
```
Comment (tmc) [inline x12_test.go]: shouldn't this include a wantErr check?

File: decode.go [hunk 2]
```diff
@@ -209,7 +209,7 @@ func scanEDI(data []byte, atEOF bool) (advance int, token []byte, err error) {
 }
 
 func parseISA(elements []string) (*ISA, error) {
-	if len(elements) < 16 {
+	if len(elements) < 17 {
 		fmt.Println("elements", elements)
```
Comment (tmc) [inline decode.go]: this print seems extraneous

**Review by tmc — COMMENTED** [review]
Thanks for adding this, do you mind addressing the touchups I mention?

File: x12_test.go (hunk 1 above)
Comment (apxamccallum) [reply x12_test.go]: I was looking for an easy way out after discovering that validate would panic when got was nil.
I suppose the better option would to check if got is nil before running validate. Honestly I haven't tried to pull TestDecode apart enough to understand everything it is trying to do, so was just doing the minimal amount of work to have the test act as expected without panics.

**Comment by apxamccallum** [comment]
Updated early return to check if error was wanted, removed the extra print, and added a couple more early returns on nil values.

**Review by tmc — APPROVED** [review]
Wonderful! Thanks.

**Pull request merged**